srcts/src/components/card.ts

```diff
@@ -74,6 +75,25 @@ class Card {
    */
   private static shinyResizeObserver = new ShinyResizeObserver();
 
+  /**
+   * Watch card parent containers for removal and exit full screen mode if a
+   * full screen card is removed from the DOM.
+   *
+   * @private
+   * @type {ShinyRemovedObserver}
+   * @static
+   */
+  private static cardRemovedObserver = new ShinyRemovedObserver(
+    `.${Card.attr.CLASS_CARD}`,
+    (el) => {
+      const card = Card.getInstance(el);
+      if (!card) return;
+      if (card.card.getAttribute(Card.attr.ATTR_FULL_SCREEN) === "true") {
+        card.exitFullScreen();
+      }
+    }
+  );
```

Comment:
Sure would be nice to this as part of a `disconnectedCallback` of a custom element instead. Is it too late (or too much work?) to make `card()` a custom element?

Reply:
I'd _like_ to make card a custom element, but I think it'll be a bit more work than we want to take on right now.

Reply:
I couldn't help but take a quick stab in https://github.com/rstudio/bslib/pull/1009 and it's actually not too bad. I think I'd prefer that we take that on after this release but before this PR

Reply:
I appreciate the enthusiasm and effort, but delaying getting this fix out there is exactly why I went this route. I prefer the custom element approach but nowhere near as much as I prefer shipping a fix. 

Unless I'm missing something, they'll be functionally equivalent, right? Which means we could take this and _then_ upgrade to a custom element in a future release.

Reply:
Yea, I see where you're coming from. If you feel strongly, I'll spend the time to review it carefully, but it is adding a fair amount of complexity for a relatively small issue. And, since we've been wanting `card()` to be a custom element, and this makes that task harder, I felt it was worth looking into before committing to the added complexity.

Reply:
> And, since we've been wanting card() to be a custom element, and this makes that task harder

I don't think this is true. I agree this approach adds more complexity, but I set it up so that we can pretty much delete the mutation observer code and keep the callback code in the card class, just moving it into the disconnected callback and removing a couple lines that wouldn't be needed in the custom element context.

Reply:
That's fair that it's a relatively minor point, but I still feel uneasy about taking on such complexity and overhead for a minor bugfix, especially when having a clear path for avoiding it.